src/server/game/Spells/Spell.cpp

```diff
@@ -3240,29 +3240,29 @@ void Spell::_cast(bool skipCheck)
 
     CallScriptBeforeCastHandlers();
 
-    // skip check if done already (for instant cast spells for example)
-    if (!skipCheck)
+    auto cleanupSpell = [this, modOwner](SpellCastResult res, uint32* p1 = nullptr, uint32* p2 = nullptr)
     {
-        auto cleanupSpell = [this, modOwner](SpellCastResult res, uint32* p1 = nullptr, uint32* p2 = nullptr)
-        {
-            SendCastResult(res, p1, p2);
-            SendInterrupted(0);
+        SendCastResult(res, p1, p2);
+        SendInterrupted(0);
 
-            if (modOwner)
-                modOwner->SetSpellModTakingSpell(this, false);
+        if (modOwner)
+            modOwner->SetSpellModTakingSpell(this, false);
 
-            finish(false);
-            SetExecutedCurrently(false);
-        };
+        finish(false);
+        SetExecutedCurrently(false);
+    };
 
-        uint32 param1 = 0, param2 = 0;
-        SpellCastResult castResult = CheckCast(false, &param1, &param2);
-        if (castResult != SPELL_CAST_OK)
-        {
-            cleanupSpell(castResult, &param1, &param2);
-            return;
-        }
+    uint32 param1 = 0, param2 = 0;
```

Comment:
why did you move this part outside of skipCheck? That is the entire point of that arg